src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -0,0 +1,88 @@
+/* ==================================================
+ * Author: Lucas Jacobs
+ *
+ * --------------------------------------------------
+ * Description:
+ *
+ * Contains functions needed to control the vision system.
+ * ================================================== */
+
+package frc.robot.subsystems;
+
+import org.opencv.core.Rect;
+import org.opencv.imgproc.Imgproc;
+
+import edu.wpi.first.cameraserver.CameraServer;
+import edu.wpi.first.cscore.UsbCamera;
+import edu.wpi.first.vision.VisionThread;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Robot;
+
+public class Vision extends SubsystemBase {
+  // HARDWARE //
+
+  // PORTS //
+
+  // CONSTANTS //
+  final static int HEIGHT = 120;
+  final static int WIDTH = 160;
+  final static int FPS = 30;
+  final static double TURN_SPEED = 0.1;
+
+  // VARIABLES //
+  private VisionThread visionThread;
+  private double xCentre;
+  private Object imgLock;
+  private Robot robot;
+
+
+  public Vision(Robot robot) {
+    this.robot = robot;
+  }
+
+  // METHODS //
+
+  /* =====================================
+  Author: Lucas Jacobs
+
+  Desc:
+  This method initalizes the camera and the vison
+  thread so that frames from the camera can be accessed
+  ===================================== */
+  public void initCamera() {
+    // initalize camera
+    UsbCamera cam = CameraServer.startAutomaticCapture();
+
+    cam.setResolution(WIDTH, HEIGHT);
+    cam.setFPS(FPS);
+
+    // initalize vision thread
+    visionThread = new VisionThread(cam, new GripPipeline(), pipeline -> {
+      
+      if (!pipeline.filterContoursOutput().isEmpty()) {
+        Rect r = Imgproc.boundingRect(pipeline.filterContoursOutput().get(0)); //Gets bounding rectangle around the target
```

Comment:
It was also nice to visualize the bounding box on the camera frame so that you can visually verify that it is working properly. Just draw a rectangle on the input frame to represent the bounding box and then output it to a camera stream so that you can see it on the dashboard.

Reply:
I've added a method that finds the biggest contour after looking at yours, and I've outputted that to the dashboard, but I haven't tested it yet. Thanks for pointing this out, I didn't think about this at all.